internal/version/release_version.go

```diff
@@ -40,15 +40,35 @@ type releaseVersionFetcher struct {
 	ctx context.Context
 }
 
-// LatestVersion encapsulates the logic to manage different cloud providers.
-func (s *releaseVersionFetcher) LatestVersion() (*ReleaseInformation, error) {
+// NVersions retrieves the first n versions returned by Github list releases endpoint.
+func (s *releaseVersionFetcher) NVersions(n int) ([]*github.RepositoryRelease, error) {
+	var releases []*github.RepositoryRelease
+	var page = 1
+	var pageSize = maxPageSize
+
+	if n <= maxPageSize {
+		pageSize = n
+	}
+
+	startTime := time.Now()
 	client := github.NewClient(nil)
-	release, _, err := client.Repositories.GetLatestRelease(s.ctx, "mongodb", "mongocli")
-	if err != nil {
-		return nil, err
+
+	for {
```

Comment:
no need for page 2, better to remove for

Reply:
note: we are assuming list releases will always return in order, in other words, most recent first

Reply:
thanks! adding all